src/shell/scripts/omp.ps1

```diff
@@ -57,11 +57,9 @@ function global:Get-PoshContext {
 }
 
 function global:Initialize-ModuleSupport {
-    if ($env:POSH_GIT_ENABLED -eq $true -and (Get-Module -Name "posh-git")) {
-        [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSProvideCommentHelp', '', Justification = 'Variable used later (not in this scope)')]
+    if ($GitPromptSettings.EnablePromptStatus) {
         $global:GitStatus = Get-GitStatus
-        [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSProvideCommentHelp', '', Justification = 'Variable used by posh-git (not in this script)')]
-        $GitPromptSettings = $null
+        $GitPromptSettings.EnablePromptStatus = $false
```

Comment:
Won't disabling this prevent all but the first `Get-GitStatus` call from returning anything?
https://github.com/dahlbyk/posh-git/blob/dc2b6367324a9db79db55186375462a454f29c2e/src/GitUtils.ps1#L247-L248

Clearing `$GitPromptSettings` will have worked differently because `Get-GitStatus` initializes default `$settings` (which would have wiped out the ability to customize `Write-GitStatus` via `$GitPromptSettings`):
https://github.com/dahlbyk/posh-git/blob/dc2b6367324a9db79db55186375462a454f29c2e/src/GitUtils.ps1#L245

Reply:
oooooooooow, I made a mistake doing this in between things. I'll fix that.

Reply:
@dahlbyk the intent was to disable the posh-git prompt from showing up. It still did that. How do I disable only that?

Reply:
We [override the prompt](https://github.com/JanDeDobbeleer/oh-my-posh/blob/749eb6cf263fd960644b9d099d973c7c94af4958/src/shell/scripts/omp.ps1#L67) just like anyone does in PowerShell. I noticed that on certain conditions, posh-git prepends itself to it.

Reply:
> Would it imply I need to advise people to initialise oh-my-posh before `Import-Module posh-git`?

That should always work. We're supposed to do nothing if `prompt` has been changed.

> I noticed that on certain conditions, posh-git prepends itself to it.

If you see that again, please check `$GitPromptSettings.AnsiConsole`. If we don't detect an ANSI-capable host, we use `Write-Host` instead of returning a string of the prompt.